lib/src/view/opening_explorer/opening_explorer_screen.dart

```diff
@@ -0,0 +1,845 @@
+                              ),
+                              child: AnalysisBoard(
+                                pgn,
+                                options,
+                                boardSize,
+                                isTablet: isTablet,
+                              ),
+                            )
+                          else
+                            AnalysisBoard(
+                              pgn,
+                              options,
+                              boardSize,
+                              isTablet: isTablet,
+                            ),
+                          Expanded(
+                            child: _OpeningExplorer(pgn: pgn, options: options),
+                          ),
+                        ],
+                      );
+              },
+            ),
+          ),
+        ),
+        _BottomBar(pgn: pgn, options: options),
+      ],
+    );
+  }
+}
+
+class _OpeningExplorer extends ConsumerWidget {
+  const _OpeningExplorer({
+    required this.pgn,
+    required this.options,
+  });
+
+  final String pgn;
+  final AnalysisOptions options;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final ctrlProvider = analysisControllerProvider(pgn, options);
+    final position = ref.watch(ctrlProvider.select((value) => value.position));
+
+    if (position.fullmoves > 24) {
+      return const Align(
+        alignment: Alignment.center,
+        child: Text('Max depth reached'),
+      );
+    }
+
+    final prefs = ref.watch(openingExplorerPreferencesProvider);
+    if (prefs.db == OpeningDatabase.player &&
+        prefs.playerDb.usernameOrId == null) {
+      return const Align(
+        alignment: Alignment.center,
+        child: Text('Select a Lichess player in the settings'),
+      );
+    }
+
+    final isRootNode = ref.watch(
+      ctrlProvider.select((s) => s.currentNode.isRoot),
+    );
+    final nodeOpening =
+        ref.watch(ctrlProvider.select((s) => s.currentNode.opening));
+    final branchOpening =
+        ref.watch(ctrlProvider.select((s) => s.currentBranchOpening));
+    final contextOpening =
+        ref.watch(ctrlProvider.select((s) => s.contextOpening));
+    final opening = isRootNode
+        ? LightOpening(
+            eco: '',
+            name: context.l10n.startPosition,
+          )
+        : nodeOpening ?? branchOpening ?? contextOpening;
+
+    final openingExplorerAsync = ref.watch(
+      openingExplorerProvider(
+        fen: position.fen,
+      ),
+    );
+
+    return openingExplorerAsync.when(
+      data: (openingExplorer) {
+        return Column(
+          children: [
+            if (openingExplorer.moves.isEmpty)
+              const Expanded(
+                child: Align(
+                  alignment: Alignment.center,
+                  child: Text('No game found'),
+                ),
+              )
+            else
+              Expanded(
+                child: SingleChildScrollView(
```

Comment:
The `SingleChildScrollView` might cause performance issues but let's leave it for now.